WcaOnRails/app/webpacker/components/Panel/Board/RegionManager.jsx

```diff
@@ -0,0 +1,170 @@
+import React from 'react';
+import {
+  Button, ButtonGroup, Confirm, Form, Header, List, Modal,
+} from 'semantic-ui-react';
+import useLoadedData from '../../../lib/hooks/useLoadedData';
+import { fetchUserGroupsUrl, addUserGroupsUrl, userGroupsUpdateUrl } from '../../../lib/requests/routes.js.erb';
+import Errored from '../../Requests/Errored';
+import Loading from '../../Requests/Loading';
+import useSaveAction from '../../../lib/hooks/useSaveAction';
+
+const defaultRegion = {
+  name: '',
+  group_type: 'delegate_regions',
+  parent_group_id: null,
+  is_active: true,
+  is_hidden: false,
+};
+
+function UserGroupVisibility({ userGroup, save, sync }) {
+  const [open, setOpen] = React.useState(false);
+  const iconName = userGroup.is_active ? 'eye' : 'eye slash';
+  return (
+    <>
+      <List.Icon
+        name={iconName}
+        link
+        onClick={() => setOpen(true)}
+      />
+      <Confirm
+        open={open}
+        content={`Are you sure you want to ${userGroup.is_active ? 'deactivate' : 'activate'} ${userGroup.name}?`}
+        onCancel={() => setOpen(false)}
+        onConfirm={() => {
+          setOpen(false);
+          save(
+            userGroupsUpdateUrl(userGroup.id),
+            { is_active: !userGroup.is_active, is_hidden: false },
```

Comment:
Why are you assuming that a deactivated group is always un-hidden?
In other words, why do you hard-code `isHidden` to `false` when changing the "active-ness"?

Reply:
Oops my bad, changed it 🙂